src/System.Linq/tests/ToHashSetTests.cs

```diff
@@ -58,17 +87,190 @@ public void TolerateDuplicates()
             // By the same token, since the normal behaviour of HashSet<T>.Add()
             // is to signal duplicates without an exception ToHashSet should
             // tolerate duplicates.
-            var hs = Enumerable.Range(0, 50).Select(i => i / 5).ToHashSet();
+            var hs = Range(0, 50).Select(i => i / 5).ToHashSet();
 
             // The OrderBy isn't strictly necessary, but that depends upon an
             // implementation detail of HashSet, so explicitly force ordering.
-            Assert.Equal(Enumerable.Range(0, 10), hs.OrderBy(i => i));
+            Assert.Equal(Range(0, 10), hs.OrderBy(i => i));
         }
 
         [Fact]
         public void ThrowOnNullSource()
         {
             Assert.Throws<ArgumentNullException>(() => ((IEnumerable<object>)null).ToHashSet());
         }
+
+        [Fact]
+        public void Distinct_WithComparer_ShouldConsiderComparer_Regression()
```

Comment:
Nit: This is the only method with underscores in its name. It would be nice to remove them to keep things consistent. (If you do so, then while you're at it you should also consider renaming `ShouldConsiderComparer` -> `ShouldConsiderIteratorComparer`.